src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1505,6 +1504,14 @@ public Set<AccessFlag> accessFlags() {
                                             AccessFlag.Location.CLASS);
     }
 
+    /**
+     * {@return true if this class has the requested {@link AccessFlag}}
+     * @param flag an {@link AccessFlag}
+     */
+    public boolean is(AccessFlag flag) {
```

Comment:
Is this API needed?   We can do `Class.accessFlags().contains(flag)`.   This discussion should belong to PR 7445 (see my comment in the `Modifier`).

Reply:
There are significant performance concerns. The current implementation of getModifers() is an intrinsic and all of the tests for modifiers are boolean operations against static constants.
Doing a Set.contains against a set that is re-built every time getAccessFlags() is called is a non-starter.
Many of the access flags have dedicated methods in j.l.Class, for example, isPublic, isStatic, etc.
It would improve usability and performance to be able to check for an accessFlag in a simple direct operation.
I'll keep this for the time being.

Reply:
This new API requires discussion and also whether it needs to be performant.  Can you bring this discussion to PR 7445.

`Class::isValue`, `Class::isIdentity` and `Class::isPrimitive` is performance-sensitive and the current implementation is performant checking against the modifiers.   Is there a case that `Class::isXXX`  cannot be used?